lib/commanded/serialization/json_serializer.ex

```diff
@@ -10,28 +10,45 @@ if Code.ensure_loaded?(Jason) do
     @doc """
     Serialize given term to JSON binary data.
     """
-    def serialize(term) do
-      Jason.encode!(term)
+    def serialize(term, config \\ []) do
+      options = maybe_options(config)
+      Jason.encode!(term, options)
     end
 
     @doc """
     Deserialize given JSON binary data to the expected type.
     """
     def deserialize(binary, config \\ []) do
-      {type, opts} =
-        case Keyword.get(config, :type) do
-          nil -> {nil, %{}}
-          type -> {TypeProvider.to_struct(type), %{keys: :atoms}}
+      type = maybe_type(config)
+
+      options =
```

Comment:
What do you think about use `if` instead of `case` for cases in what is binary?
```suggestion
      options = if type, do? Map.merge(%{keys: :atoms}, maybe_options(config)), else: %{}
```